src/services/shapeService.utils.ts

```diff
@@ -36,107 +48,176 @@ export const parseSingleShape = (item: string): Promise<Quad[]> => {
+    if (parameter.name === PARAM_IN) {
+      const firstItemId = parameter.value;
+      value = getInPropertyValues(firstItemId, linkedListValues);
     }
+    return { type, value, } as PropertyValue
   });
 
-  propertiesMap.forEach(properties => {
-    properties.forEach(property => {
-      property.propertyValues.sort((a, b) => {
-        if (a.type === 'http://www.w3.org/ns/shacl#name') {return -1;}
-        if (b.type === 'http://www.w3.org/ns/shacl#name') {return 1;}
-        if (a.type === 'http://www.w3.org/ns/shacl#description') {return -1;}
-        if (b.type === 'http://www.w3.org/ns/shacl#description') {return 1;}
-        return a.type.localeCompare(b.type);
-      });
-    });
-  })
-
-  shapesMap.forEach(shape => {
-    shape.propertyIds.forEach(propertyId => {
-      shape.properties.push(...(propertiesMap.get(propertyId) || []));
-      shape.nodes.push(...(nodesMap.get(propertyId) || []));
-    });
+  return params.sort((a, b) => {
+    const indexA = PARAMETER_TYPES.indexOf(a.type);
+    const indexB = PARAMETER_TYPES.indexOf(b.type);
+    return indexA - indexB;
   });
+}
 
-  return Array.from(shapesMap.values());
-};
+const getShapeProperties = (shapeId: string, propertyIds: Map<string, string[]>, parameters: Map<string, Parameter[]>, linkedListValues: Map<string, ListItem>) => {
+  const shapePropertyIds = propertyIds.get(shapeId) || [];
+  return shapePropertyIds.map(propertyId => {
+    const propertyValues = getPropertyValues(propertyId, parameters, linkedListValues);
+    const propertyPath = propertyValues.find(prop => prop.type === PARAM_PATH);
 
-export const getShapeByName = async (name: string): Promise<Shape | undefined> => {
-  const schemas = await fetchAllSchemas();
-  const allShapes = await fetchAllShapesFromSchemas(schemas);
-  return allShapes ? allShapes.find(shape => shape.subject === name) : undefined;
+    return {
+      propertyId: propertyPath ? propertyPath.value : propertyId,
+      propertyValues
+    }
+  });
+}
+
+const getNodes = (shapeId: string, propertyIds: Map<string, string[]>, nodeIds: Map<string, string>): string[] => {
+  const shapeProperties = propertyIds.get(shapeId) || [];
+  return shapeProperties
+    .map(propertyId => nodeIds.get(propertyId))
+    .filter(nodeId => nodeId !== undefined) as string[];
 }
 
-export const findRelatedShapes = (shapes: Shape[], ontologyId: string): Shape[] => {
-  return shapes.filter(
-    shape => shape.targetClasses.some(
-      targetClasses => targetClasses.includes(ontologyId)));
+export const createShapeObjects = (shaclShapeId: string, quads: Quad[]): Shape[] => {
+  const linkedListValues = new Map<string, ListItem>();
+  const parameters = new Map<string, Parameter[]>();
+  const nodeIds = new Map<string, string>();
+  const propertyIds = new Map<string, string[]>();
+  const targetClasses = new Map<string, string[]>();
+  const shaclShapeNames = [] as string[];
+
+  quads.forEach(quad => {
```

Comment:
Only one If statement is executed at a time. It should then not be necessary to test further If statements. Wouldn't if else or switch be better here?

Reply:
It is defenetly a good ideea. I would prefer **else if** statements because having well named check methods are more expressive than having a bunch of URLs in a swich statements. The URLs are details that disrupt the understanding of the algorithm.

done